x-pack/plugin/sql/qa/jdbc/src/main/java/org/elasticsearch/xpack/sql/qa/jdbc/ResultSetTestCase.java

```diff
@@ -2244,6 +2245,30 @@ public void testResultSetConsumed() throws IOException {
         assertEquals("No row available", sqle.getMessage());
     }
 
+    public void testSingleVersionFieldValue() throws SQLException, IOException {
+        assumeTrue("Driver version [" + JDBC_DRIVER_VERSION + "] doesn't support VERSION fields", isVersionFieldTypeSupported());
+
+        createTestDataForVersionType();
+        String query = "SELECT name, version from test where version = '1.3.0'";
+        doWithQuery(query, results -> {
+            results.next();
+            assertEquals("version 1.3.0", results.getString("name"));
+            assertEquals("1.3.0", results.getString("version"));
+            SQLException sqle = expectThrows(SQLException.class, () -> results.getByte("version"));
+            assertEquals(format(Locale.ROOT, "Unable to convert value [%.128s] of type [VERSION] to [Byte]", "1.3.0"), sqle.getMessage());
+            assertFalse(results.next());
+        });
+
+        // bad version value
+        query = "SELECT name, version from test where version = 'foo'";
+        doWithQuery(query, results -> {
+            results.next();
+            assertEquals("version foo", results.getString("name"));
+            assertEquals("foo", results.getString("version"));
+            assertFalse(results.next());
+        });
+    }
```

Comment:
Since there is this distinction between valid and invalid versions, I thought it would be good to have both in the result set test. Anyway, I agree that knowing the implementation details this test is probably not crucial. I'll refactor it a bit to make it shorter.